### PR TITLE
Use Dynawo 1.4.0 and fix DynaFlow integration

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Bug fix

**What is the new behavior (if this is a feature change)?**
This PR fixes many issues:
 - Dynawo side SlackTerminal extension parsing fixed in 1.4.0 (1.3.1 is buggy)
 - IIDM XML is exported with unsupported (from IIDM C++ side) extensions => now we explicitely defined list of written extensions.
 - IIDM export version and extensions should be common to DynaFlow and DynaWaltz (because provided by Dynawo platform) 
 - Dynaflow security analysis was still exporting IIDM XML to 1.2 => now 1.4, many reference files have been updated
 - DynaFlow output files were written in the wrong directory, local one instead of remote one. The bug is not visible using the local computation manager but is when using Docker/Slurm/Mpi ones where the 2 directories are not the same.
 - DynaWaltz version is wrong.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by geofjamg** [comment]
@flo-dup code coverage is low because a distribution module is missing to get cross module coverage.

File: dynawaltz/src/main/java/com/powsybl/dynawaltz/DynaWaltzProvider.java [hunk 1]
```diff
@@ -75,7 +74,7 @@ public String getName() {
 
     @Override
     public String getVersion() {
-        return "1.2.0";
+        return VERSION;
```
Comment (flo-dup) [inline dynawaltz/src/main/java/com/powsybl/dynawaltz/DynaWaltzProvider.java:77]: In another implementation, the version number is not the binary version number but the adapter version. Shouldn't it be the same here? We should do this using `AbstractVersion` from `powsybl-tools`

Comment (geofjamg) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/DynaWaltzProvider.java:77]: I would say yes to stay simple we can just be sync with dynamo version

Comment (flo-dup) [reply dynawaltz/src/main/java/com/powsybl/dynawaltz/DynaWaltzProvider.java:77]: Well, it will always return 1.4.0 now, even if it's the 1.4.1 installed

**Review by flo-dup — APPROVED** [review]
We'll fix (or not) the providers version in another PR

**Pull request merged**